add some more curl examples and document the query parameter controls on HEC compatible ingester (#824)
gravwell/wiki: ingesters/http.md

--data-binary @- -H "Content-Encoding: gzip"
```

#### Sending Multiple Entries

> Each of the HEC endpoints support sending mutiple entries in a single request; batching up multiple entries in a single request is dramatically more efficient than one entry per request.

Reviewer: ```suggestion
Each of the HEC endpoints support sending multiple entries in a single request; batching up multiple entries in a single request is dramatically more efficient than one entry per request.
```

> The structured data endpoints allow for specifying a sourcetype value which will be applied to the `Tag-Match` config parameter, if a match is found a new tag is applied.  The raw endpoints can provide a sourcetype as a query parameter, or a direct tag value can be provided to bypass the sourcetype translations entirely and explicitely specify the tag.

Reviewer: ```suggestion
The structured data endpoints allow for specifying a sourcetype value which will be applied to the `Tag-Match` config parameter, if a match is found a new tag is applied.  The raw endpoints can provide a sourcetype as a query parameter, or a direct tag value can be provided to bypass the sourcetype translations entirely and explicitly specify the tag.
```

> The structured data endpoints allow for specifying a sourcetype value which will be applied to the `Tag-Match` config parameter, if a match is found a new tag is applied.  The raw endpoints can provide a sourcetype as a query parameter, or a direct tag value can be provided to bypass the sourcetype translations entirely and explicitly specify the tag.

Reviewer 2: ```suggestion
The structured data endpoints allow for specifying a sourcetype value which will be applied to the `Tag-Match` config parameter; if a match is found, a new tag is applied.  The raw endpoints can provide a sourcetype as a query parameter, or a direct tag value can be provided to bypass the sourcetype translations entirely and explicitly specify the tag.
```

> This curl command shows shows sending several structured entries to the structured endpoint:

Reviewer: ```suggestion
This curl command shows sending several structured entries to the structured endpoint:
```

```
curl --http1.1 -X POST -v http://example.gravwell.io/services/collector/event \
    -H "Authorization: Splunk thisisyourtoken" -d '
    {"event": "Hello, this is a structured event"}
    {"event": "Hello, this is another structured event"}
    {"event": "Hello, this is yet another structured event"}'
```

> This curl command shows shows sending several entries to the raw endpoint:

Reviewer: ```suggestion
This curl command shows sending several entries to the raw endpoint:
```

```
curl --http1.1 -X POST -v http://example.gravwell.io/services/collector/raw \
    -H "Authorization: Splunk thisisyourtoken" -d '
    hello, welcome to raw #1
    hello, welcome to raw #2
    hello, welcome to raw #3'
```

> This curl command shows sending several entries to the raw endpoint and providing a sourcetype and tag as a query parameter.

Reviewer 2: ```suggestion
The following curl commands show sending several entries to the raw endpoint and providing a sourcetype and tag as a query parameter.
```

> This curl command shows shows sending several entries to the raw endpoint and providing a sourcetype and tag as a query parameter.

Reviewer: ```suggestion
This curl command shows sending several entries to the raw endpoint and providing a sourcetype and tag as a query parameter.
```

> Override the sourcetype to use the Tag-Match parameter.

Reviewer 2: ```suggestion
Override the sourcetype to use the Tag-Match parameter:
```

> Override the sourcetype to use the Tag-Match parameter

Reviewer: ```suggestion
Override the sourcetype to use the Tag-Match parameter.
```

```
curl --http1.1 -X POST -v http://example.gravwell.io/services/collector/raw?sourcetype=foobar \
    -H "Authorization: Splunk thisisyourtoken" -d '
    hello, welcome to raw #1
    hello, welcome to raw #2
    hello, welcome to raw #3'
```

> Override the tag value directly.

Reviewer 2: ```suggestion
Override the tag value directly:
```

> Override the tag value directly

Reviewer: ```suggestion
Override the tag value directly.
```

```
curl --http1.1 -X POST -v http://example.gravwell.io/services/collector/raw?tag=testing \
    -H "Authorization: Splunk thisisyourtoken" -d '
    hello, welcome to raw #1
    hello, welcome to raw #2
    hello, welcome to raw #3'
```

> This curl command shows shows sending several entries to the structured endpoint with a default tag and setting specific entry sourcetypes:

Reviewer: ```suggestion
This curl command shows sending several entries to the structured endpoint with a default tag and setting specific entry sourcetypes:
```

Some characters that are supported in a Splunk sourcetype are not supported in a Gravwell tag. If you need to specify a sourcetype with special characters, surround the Tag-Match argument in backticks to specify a raw string and surround the sourcetype in double quotes.
```

#### Debug-Posts

> The `Debug-Posts` configuration option allows for gathering additional data on each HTTP POST request to the HTTP ingester endpoint.  Only successful transactions will be logged when using the `Debug-Posts` configuration option, authentication failures, structure failures, or just bad requests are logged using the existing systems.  The bebug logs are sent to the `gravwell` tag.

Reviewer 2: ```suggestion
The `Debug-Posts` configuration option allows for gathering additional data on each HTTP POST request to the HTTP ingester endpoint.  Only successful transactions will be logged when using the `Debug-Posts` configuration option.  Authentication failures, structure failures, or just bad requests are logged using the existing systems.  The debug logs are sent to the `gravwell` tag.
```
